crates/doc/src/ptr.rs

```diff
@@ -140,12 +146,13 @@ impl Pointer {
                 Node::Object(fields) => match token {
                     Token::Index(ind) => fields.get(&ind.to_string()),
                     Token::Property(property) => fields.get(&property),
+                    Token::AdditionalProperties => None,
```

Comment:
🤔 it looks like `Token::from_str` does parse `-` into `NextIndex`. I'm thinking it shouldn't, which matches how `NextProperty` is being implemented. Then it should return `None` here as well, because you can never really query `NextIndex` or `NextProperty`.